ssh/session/session.go

```diff
@@ -204,6 +257,38 @@ func NewSession(ctx gliderssh.Context, tunnel *httptunnel.Tunnel, cache cache.Ca
 	return session, nil
 }
 
+func (s *Session) NewClientChannel(newChannel gossh.NewChannel, seat int) (*ClientChannel, error) {
+	channel, requests, err := newChannel.Accept()
+	if err != nil {
+		return nil, err
+	}
+
+	c := &ClientChannel{
+		Channel:  channel,
+		Requests: requests,
+	}
+
+	s.Client.Channels[seat] = c
+
+	return c, nil
+}
+
+func (s *Session) NewAgentChannel(name string, seat int) (*AgentChannel, error) {
+	channel, requests, err := s.Agent.Client.OpenChannel(name, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	a := &AgentChannel{
+		Channel:  channel,
+		Requests: requests,
+	}
+
+	s.Agent.Channels[seat] = a
```

Comment:
What happens if I call that twice for the same seat ID?

Reply:
For these cases, I've added an error to handle it.